manifests/master.pp

```diff
@@ -79,8 +80,13 @@
   if $::osfamily != 'RedHat' {
     fail("Cannot configure an IPA master server on ${::operatingsystem} operating systems. Must be a RedHat-like operating system.")
   }
+  if $dns == true {
+    $pkglist=[$ipa::master::svrdnspkg,$ipa::master::svrpkg]
+  } else {
+    $pkglist=$ipa::master::svrpkg
+  }
 
-  realize Package[$ipa::master::svrpkg]
+  realize Package[$pkglist]
```

Comment:
It will only realize once:

> The realize function can be used multiple times on the same virtual resource and the resource is only managed once.

https://docs.puppet.com/puppet/latest/reference/lang_virtual.html#realizing-with-the-realize-function